types/src/proof/definition.rs

```diff
@@ -844,20 +842,72 @@ impl TransactionListProof {
             })
             .collect::<Result<Vec<_>>>()?;
 
-        let txn_info_hashes: Vec<_> = self
-            .transaction_infos
-            .iter()
-            .map(CryptoHash::hash)
-            .collect();
-        self.ledger_info_to_transaction_infos_proof.verify(
-            ledger_info.transaction_accumulator_hash(),
+        verify_transaction_infos_against_ledger_info(
+            &self.transaction_infos,
             first_transaction_version,
-            &txn_info_hashes,
-        )?;
-        Ok(())
+            ledger_info,
+            &self.ledger_info_to_transaction_infos_proof,
+        )
     }
 }
 
+/// The proof used to authenticate a list of consecutive transaction infos.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub struct TransactionInfoListProof {
```

Comment:
This is exactly the same with `TransactionListProof`?

And assuming there's not that, it sticks more to our naming convention is called `TransactionInfoListWithProof`.

Reply:
yeah, I'd prefer `TransactionInfoListWithProof` over `TransactionListProof`. I'd also prefer only one type, rather than two : p. I'm actually not sure there's any backwards compatibility issue with just renaming `TransactionListProof`, considering it's only ever serialized with bcs and isn't hashed (as far as I know).

Reply:
The difference between (A) `TransactionInfoListProof` and (B) `TransactionListProof` is in the `verify()` method:
- (A) only verifies the transaction info list is attested to by the given ledger info.
- (B) verifies the same as (A) but also checks that the transaction hashes themselves match those specified by the transaction infos.

1. I went with `TransactionInfoListProof` to be consistent with `TransactionListProof`, but I can change it to `TransactionInfoListWithProof`. This makes `TransactionListProof` the odd one out 😄
2. I'd also prefer no duplication, but honestly, I'm not super confident. @phlip9: how confident are you that this rename won't break something in the eco-system? 😄 Looking briefly at the BCS spec, I see that the name shouldn't ever be used... but I'm still on the fence :P